packages/react-reconciler/src/__tests__/ReactNoopRendererAct-test.js

```diff
@@ -0,0 +1,63 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * @jest-environment node
+ */
+
+// sanity tests for ReactNoop.act()
+
+jest.useRealTimers();
+const React = require('react');
+const ReactNoop = require('react-noop-renderer');
+const Scheduler = require('scheduler');
+
+describe('ReactNoop.act()', () => {
```

Comment:
Because the macrotask thing is so subtle, we need tests that correspond each of the platforms where you might run tests.

- Node + jsdom (or other fake dom)
- Node without jsdom
- React Native
- No setImmediate
- Polyfilled setImmediate

Reply:
Polyfilling setImmediate is very common and the semantics are non-standard. It's supposed to be a macrotask but I don't know if we can rely on that.

Reply:
For example, this part indicates to me that `act` will fail if neither `setImmediate` nor `MessageChannel` is defined:

https://github.com/facebook/react/pull/14853/files#r265820721

(Maybe I'm missing something, but regardless, it would be nice to have a test that verifies this.)

Reply:
> we need tests that correspond to each of the platforms where you might run tests.

do you still feel strongly about this? now that - 
- we use the non-mocked setImmediate from node
- we aren't trying to use anything beyond MessageChannel (and warn if you try in a browser that doesn't have it)

Reply:
They don't need to be super elaborate tests, just a sanity check that this won't insta-fail in those environments so that the next time someone touches this code they don't forget to consider those cases.

Reply:
btw, the decision was to let it insta fail when setImmediate/MessageChannel aren't available (so, 'old' IE). I have a backup plan in case that's needed, but letting it be. I'll set up tests for the rest.

Reply:
I'm personally fine with that plan so long as there's a useful error message that explains what's going on clearly please. Because I just know I'll get issues on react-testing-library otherwise.

Reply:
https://github.com/facebook/react/pull/14853/files#diff-08838400dbefc5107cea81326e3d9847R176

Reply:
Perfect 💯